### PR TITLE
Marked some unlock functions noexcept

### PR DESCRIPTION
**Pull request by goedderz**
I'd like at least two reviews for this PR!

### Scope & Purpose

This started out by marking `~WriteLocker()` noexcept, and making necessary changes down the line. I also made the (un)lock methods of `ReadWriteSpinLock`, which can fail in theory, `[[nodiscard]]`. There were several places relying on them to work eventually, so I provided appropriate methods that will not fail.

- [X] Bug-Fix and cleanup for *devel-branch*
- [X] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7316/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7341/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7346/

**Comment by goedderz** [comment]
Last Jenkins is blue.

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7646/

**Comment by goedderz** [comment]
Jenkins is blue.

File: arangod/Cache/Manager.h [hunk 1]
```diff
@@ -36,9 +36,10 @@
...
 #include "Cache/TransactionManager.h"
 
-#include <stdint.h>
+#include <array>
 #include <atomic>
 #include <chrono>
+#include <cstdint>
 #include <map>
 #include <memory>
 #include <stack>
```
Comment (jsteemann) [inline arangod/Cache/Manager.h]: this line can be removed now, right?
```suggestion
```

Comment (goedderz) [reply arangod/Cache/Manager.h]: No, `std::stack` is still in use. I just replaced a C-Array of stacks with an `std::array` of stacks.

Comment (jsteemann) [reply arangod/Cache/Manager.h]: ok

**Pull request merged**